### PR TITLE
Between operator for segments with numeric Custom Object fields

### PR DESCRIPTION
**Pull request by aarohiprasad**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | [ No]
| New feature/enhancement? (use the a.x branch)      | [ Yes]
| Deprecations?                          | [ NA]
| BC breaks? (use the c.x branch)        | [NA ]
| Automated tests included?              | [ Yes] 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:

This PR implements the between operator in segments for numeric fields.

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create a Custom Object People/Person and add a number field "Family Members".
3. Create 3 custom items from this object; 'A', 'B' and 'C' and assign them 3, 7 and 12 family members respectively.
4. Create 3 contacts 'A', 'B' and 'C' and link them to People 'A', 'B' and 'C' respectively.
5. Create a new segment "Family Members Between Segment" and add People:Family Members as filter field.
6. Choose "between" operator. Two input fields should appear. 
image

7. Enter 5 and 10 in the "From" and "To" boxes respectively and save the segment.
8. Wait for the segment to build.
9. Verify that only contact B (the one with 7 family members) is a member of the segment.
10. Additionally try saving the segment leaving either of the two input fields blank, or entering non integer values. You should see an error message. 
image
11. Create another segment, "Family Members Not Between Segment", with same configurations as the previous one. Except this time choose "not between" operator.

12. Wait for the segment to build and verify that this time the contacts with 3 and 12 family members are members of the segment.

### CODE REVIEW

File: Helper/QueryFilterHelper.php [hunk 1]
```diff
@@ -154,14 +155,15 @@ private function addOperatorExpression(
...
+     * @param float|bool|array<mixed>|string|null $filterParameterValue
      */
     private function getCustomValueValueExpression(
         SegmentQueryBuilder $customQuery,
         string $tableAlias,
         string $operator,
-        string $valueParameter
-    ) {
+        string $valueParameter,
+        float|bool|array|string $filterParameterValue = null,
```
Comment (escopecz) [inline Helper/QueryFilterHelper.php]: I'm afraid this plugin still needs to run on PHP 7.4 as defined in composer.json as someone from the community can run it on PHP 7.4 with Mautic 4. Can we remove the type here and add it as `mixed` in the comments?

Same with the return type.

**Review by escopecz — CHANGES_REQUESTED** [review]

**Review by avikarshasaha — COMMENTED** [review]
@aarohiprasad Please re-request for review when it is done.

File: Helper/QueryFilterHelper.php [hunk 2]
```diff
@@ -154,14 +155,15 @@ private function addOperatorExpression(
...
      */
     private function getCustomValueValueExpression(
         SegmentQueryBuilder $customQuery,
         string $tableAlias,
         string $operator,
-        string $valueParameter
-    ) {
+        string $valueParameter,
+        $filterParameterValue = null,
+    ): CompositeExpression|string {
```
Comment (escopecz) [inline Helper/QueryFilterHelper.php]: As I noted previously, this won't work for PHP 7.4.

**Review by escopecz — CHANGES_REQUESTED** [review]

**Review by escopecz — DISMISSED** [review]
I see no more issues, thanks!

**Comment by aarohiprasad** [comment]
Only one commit has been added since the last review. Please review [8de83d7](https://github.com/acquia/mc-cs-plugin-custom-objects/pull/346/commits/8de83d72bc11c7853292fc329e5e53c322392978). @escopecz @avikarshasaha. Thank you!

**Review by escopecz — DISMISSED** [review]

**Review by avikarshasaha — CHANGES_REQUESTED** [review]
This is not working in the beta version. @aarohiprasad is having some issues with the merging. She will fix it.

**Review by avikarshasaha — DISMISSED** [review]

**Review by escopecz — DISMISSED** [review]
reapproving

**Review by avikarshasaha — DISMISSED** [review]

**Review by avikarshasaha — APPROVED** [review]

**Review by rahuld-dev — APPROVED** [review]
approving PR after resolved conflicts

**Pull request merged**